tools/packaging/static-build/shim-v2/build.sh

```diff
@@ -16,34 +16,45 @@ VMM_CONFIGS="qemu fc"
 
 GO_VERSION=${GO_VERSION}
 RUST_VERSION=${RUST_VERSION}
+CC=""
 
 DESTDIR=${DESTDIR:-${PWD}}
 PREFIX=${PREFIX:-/opt/kata}
 container_image="${SHIM_V2_CONTAINER_BUILDER:-$(get_shim_v2_image_name)}"
 
+[ "${CROSS_BUILD}" == "true" ] && container_image_bk="${container_image}" && container_image="${container_image%x86_64}${ARCH}-cross-build"
 sudo docker pull ${container_image} || \
-       	(sudo docker build  \
+	(sudo docker ${BUILDX} build ${PLATFORM}  \
 		--build-arg GO_VERSION="${GO_VERSION}" \
 		--build-arg RUST_VERSION="${RUST_VERSION}" \
 		-t "${container_image}" \
 		"${script_dir}" && \
-	 push_to_registry "${container_image}")
+	push_to_registry "${container_image}")
 
-arch=$(uname -m)
+arch=${ARCH:-$(uname -m)}
 if [ ${arch} = "ppc64le" ]; then
 	arch="ppc64"
 fi
 
+#Build rust project using cross build musl image to speed up
+[[ "${CROSS_BUILD}" == "true" && ${ARCH} != "s390" ]] && container_image="messense/rust-musl-cross:${ARCH}-musl" && CC=${ARCH}-unknown-linux-musl-gcc
```

Comment:
@jongwu, here there should also be a check added power since musl is not supported on ppc64le.
`[[ "${CROSS_BUILD}" == "true" && ( ${ARCH} != "s390" && ${ARCH} != "ppc64le" ) ]] && ...`